synapse/replication/tcp/streams/_base.py

```diff
@@ -305,6 +305,14 @@ def minimal_local_current_token(self) -> Token:
         # which means we need to negate it.
         return -self.store._backfill_id_gen.get_minimal_local_current_token()
 
+    def can_discard_position(
+        self, instance_name: str, prev_token: int, new_token: int
+    ) -> bool:
+        # Caches streams can't go backwards, so we know we can ignore any
+        # positions where the tokens are from before the current token.
```

Comment:
This isn't the caches stream?

Reply:
Oh, I forgot I did the backfill here too